svgnative/src/SVGDocument.cpp

```diff
@@ -55,51 +55,106 @@ SVGDocument::~SVGDocument() {}
 
 void SVGDocument::Render()
 {
+    if (!mDocument)
+        return;
+
     ColorMap colorMap;
     mDocument->Render(colorMap, mDocument->mViewBox[2], mDocument->mViewBox[3]);
 }
 
 void SVGDocument::Render(float width, float height)
 {
+    if (!mDocument)
+        return;
+
     ColorMap colorMap;
     mDocument->Render(colorMap, width, height);
 }
 
 void SVGDocument::Render(const ColorMap& colorMap)
 {
+    if (!mDocument)
+        return;
+
     mDocument->Render(colorMap, mDocument->mViewBox[2], mDocument->mViewBox[3]);
 }
 
-void SVGDocument::Render(const ColorMap& colorMap, float width, float height) { mDocument->Render(colorMap, width, height); }
+void SVGDocument::Render(const ColorMap& colorMap, float width, float height) {
+    if (!mDocument)
+        return;
+
+    mDocument->Render(colorMap, width, height);
+}
 
 void SVGDocument::Render(const char* id)
 {
+    if (!mDocument)
+        return;
+
     ColorMap colorMap;
     mDocument->Render(id, colorMap, mDocument->mViewBox[2], mDocument->mViewBox[3]);
 }
 
 void SVGDocument::Render(const char* id, float width, float height)
 {
+    if (!mDocument)
+        return;
+
     ColorMap colorMap;
     mDocument->Render(id, colorMap, width, height);
 }
 
 void SVGDocument::Render(const char* id, const ColorMap& colorMap)
 {
+    if (!mDocument)
+        return;
+
     mDocument->Render(id, colorMap, mDocument->mViewBox[2], mDocument->mViewBox[3]);
 }
 
-void SVGDocument::Render(const char* id, const ColorMap& colorMap, float width, float height) { mDocument->Render(id, colorMap, width, height); }
+void SVGDocument::Render(const char* id, const ColorMap& colorMap, float width, float height)
+{
+    if (!mDocument)
+        return;
+
+    mDocument->Render(id, colorMap, width, height);
+}
+
+std::int32_t SVGDocument::Width() const
+{
+    return static_cast<std::int32_t>(mDocument->mViewBox[2]);
```

Comment:
Isn't the mDocument check missing here too?

Reply:
Indeed, it was my mistake. Sorry, I added it.